<img src="https://avatars.githubusercontent.com/u/16632335?v=4" alt="Avatar" style="border-radius: 50%; width: 150px; height: 150px;">

# By Raul Rodrigues - HSR

## 1. Importanto a biblioteca e lendo as tabelas
> Segue os códigos abaixo, qualquer dúvida, fico a disposição, desde já obrigado, e tenha uma ótima semana!

> Toda resposta eu criei um breve insight de forma superficial, por não ter mais informações do negócio em si, com essa informação mais precisa, mais insights profundos poderiam ser elaborados

In [1]:
import pandas as pd

In [2]:
painelistas = pd.read_csv("Painelista.csv", sep=';')

# Imprimindo o shape e o nome das colunas do respectivo DataFrame
print("Shape:", painelistas.shape)
# Mostrando o nome das colunas
print("\nNome das colunas:", painelistas.columns)
# Mostrando os tipos das colunas
print("\nTipos das colunas:")
print(painelistas.dtypes)

Shape: (600, 5)

Nome das colunas: Index(['PainelistaID', 'Cidade', 'Estado', 'Sexo', 'Nascimento'], dtype='object')

Tipos das colunas:
PainelistaID    object
Cidade          object
Estado          object
Sexo            object
Nascimento      object
dtype: object


In [3]:
pesquisa = pd.read_csv("Pesquisa.csv", sep=';')

# Imprimindo o shape e o nome das colunas do respectivo DataFrame
print("Shape:", pesquisa.shape)
# Mostrando o nome das colunas
print("\nNome das colunas:", pesquisa.columns)
# Mostrando os tipos das colunas
print("\nTipos das colunas:")
print(pesquisa.dtypes)

Shape: (2, 3)

Nome das colunas: Index(['PesquisaID', 'Nome', 'CodJob'], dtype='object')

Tipos das colunas:
PesquisaID    object
Nome          object
CodJob         int64
dtype: object


In [4]:
painelista_pesquisa = pd.read_csv("Painelista_Pesquisa.csv", sep=';')

# Imprimindo o shape e o nome das colunas do respectivo DataFrame
print("Shape:", painelista_pesquisa.shape)
# Mostrando o nome das colunas
print("\nNome das colunas:", painelista_pesquisa.columns)
# Mostrando os tipos das colunas
print("\nTipos das colunas:")
print(painelista_pesquisa.dtypes)

Shape: (1200, 4)

Nome das colunas: Index(['PesquisaID', 'PainelistaID', 'Data', 'CodCliente'], dtype='object')

Tipos das colunas:
PesquisaID       object
PainelistaID     object
Data             object
CodCliente      float64
dtype: object


In [5]:
respostas = pd.read_csv("Respostas.csv", sep=';')

# Imprimindo o shape e o nome das colunas do respectivo DataFrame
print("Shape:", respostas.shape)
# Mostrando o nome das colunas
print("\nNome das colunas:", respostas.columns)
# Mostrando os tipos das colunas
print("\nTipos das colunas:")
print(respostas.dtypes)

Shape: (4382, 5)

Nome das colunas: Index(['CodCliente', 'CodJob', 'CodQuestao', 'DescQuestaoPOR', 'Resposta'], dtype='object')

Tipos das colunas:
CodCliente         int64
CodJob             int64
CodQuestao         int64
DescQuestaoPOR    object
Resposta          object
dtype: object


In [6]:
# Carregando o arquivo CSV em um DataFrame
questoes_respostas = pd.read_csv("Questao_DescResposta.csv", sep=';')

# Imprimindo o shape e o nome das colunas do respectivo DataFrame
print("Shape:", questoes_respostas.shape)
# Mostrando o nome das colunas
print("\nNome das colunas:", questoes_respostas.columns)
# Mostrando os tipos das colunas
print("\nTipos das colunas:")
print(questoes_respostas.dtypes)

Shape: (92, 3)

Nome das colunas: Index(['CodQuestao', 'Resposta', 'DescOpcaoPOR'], dtype='object')

Tipos das colunas:
CodQuestao       int64
Resposta         int64
DescOpcaoPOR    object
dtype: object


## 2. O MERGE das tabelelas, será feito na seguinte ordem:

- 1. respostas com pesquisa
- 2. respostas(merged_df) com questoes_respostas
- 3. painelistas com painelista_pesquisa
- 4. pesquisa com painelista_pesquisa(merged_03) 
- 5. resposta(merged_df) com painelista_pesquisa(merged_04)

### 2.1 Tratamento da coluna Resposta da coluna Respostas:
Antes de começar a realizar os merges é necessário realizar o tratamento da coluna Resposta da tabela Respostas, pois tem perguntas que tem mais de uma resposta do mesmo cliente.

In [7]:
# Linhas com resposta única
df_expandido = respostas[~respostas['Resposta'].str.contains("\|")]

# Linhas com respostas múltiplas
df_multiplo = respostas[respostas['Resposta'].str.contains("\|")]

# Processar linhas com respostas múltiplas
for index, row in df_multiplo.iterrows():
    respostas_lista = row['Resposta'].split("|")
    for resposta in respostas_lista:
        nova_linha = row.copy()
        nova_linha['Resposta'] = int(resposta)
        df_expandido = pd.concat([df_expandido, pd.DataFrame([nova_linha])], ignore_index=True)

# Visualizar as primeiras linhas do resultado
df_expandido.head()

CodCliente  CodJob  CodQuestao  DescQuestaoPOR Resposta
0    17529748    4037      122498  Você tem PIX?         3
1    17529787    4037      122498  Você tem PIX?         1
2    17662142    4037      122498  Você tem PIX?         1
3    17662151    4037      122498  Você tem PIX?         1
4    17689472    4037      122498  Você tem PIX?         1

In [8]:
df_expandido.shape

(5551, 5)

### 2.2 Merge 01: respostas(df_expandido) com pesquisa na chave CodJob

In [9]:
merged_df = pd.merge(df_expandido, pesquisa, on='CodJob', how='left')
merged_df.shape

(5551, 7)

In [10]:
merged_df.head()

CodCliente  CodJob  CodQuestao  DescQuestaoPOR Resposta  \
0    17529748    4037      122498  Você tem PIX?         3   
1    17529787    4037      122498  Você tem PIX?         1   
2    17662142    4037      122498  Você tem PIX?         1   
3    17662151    4037      122498  Você tem PIX?         1   
4    17689472    4037      122498  Você tem PIX?         1   

                             PesquisaID     Nome  
0  31821770-8E76-434D-BB14-10A6CA8B7667  98DZK9M  
1  31821770-8E76-434D-BB14-10A6CA8B7667  98DZK9M  
2  31821770-8E76-434D-BB14-10A6CA8B7667  98DZK9M  
3  31821770-8E76-434D-BB14-10A6CA8B7667  98DZK9M  
4  31821770-8E76-434D-BB14-10A6CA8B7667  98DZK9M

> Foi tentado realizar o merge entre respostas com pesquisa na CodCliente e Resposta: merged_df = pd.merge(merged_df, questoes_respostas, on=['CodQuestao', 'Resposta'], how='left')

> Porém retornou erro de que estava tentando dar um merge com tipos diferentes das colunas, portanto foi optado em transformar tudo para string, por ter numeros e textos, depois fazer a conversar necessária.

In [11]:
# Convertendo a coluna 'CodQuestao' para o tipo int
merged_df['CodQuestao'] = merged_df['CodQuestao'].astype(int)
questoes_respostas['CodQuestao'] = questoes_respostas['CodQuestao'].astype(int)


# Convertendo a coluna 'Resposta' para o tipo string
merged_df['Resposta'] = merged_df['Resposta'].astype(int)
questoes_respostas['Resposta'] = questoes_respostas['Resposta'].astype(int)

### 2.3 Merge 02: respostas(merged_df) com questoes_respostas na CodCliente

In [12]:
merged_df = pd.merge(merged_df, questoes_respostas, on=['CodQuestao', 'Resposta'], how='inner')
merged_df.shape

(5551, 8)

### 2.4 Merge 03: painelistas com painelista_pesquisa na chave PainelistaID

In [13]:
merged_03 = pd.merge(painelistas, painelista_pesquisa, on='PainelistaID', how='left')
merged_03.shape

(1200, 8)

In [14]:
merged_03.head()

PainelistaID           Cidade Estado Sexo  \
0  44C12860-704A-4BFC-AA76-46BD5604B60B       Uberlândia     MG    M   
1  44C12860-704A-4BFC-AA76-46BD5604B60B       Uberlândia     MG    M   
2  46A9177F-BFC7-44A2-8572-545C76FC38E3          Itapeva     SP    F   
3  46A9177F-BFC7-44A2-8572-545C76FC38E3          Itapeva     SP    F   
4  1E69D5A1-91BC-4966-A2C0-FFB6C3CF52D8  Taboão da Serra     SP    F   

  Nascimento                            PesquisaID              Data  \
0    09/1994  D8811E98-FE36-4575-99E5-68BE7D31F6FB  14/03/2022 17:40   
1    09/1994  31821770-8E76-434D-BB14-10A6CA8B7667  14/03/2022 17:48   
2    05/1999  D8811E98-FE36-4575-99E5-68BE7D31F6FB  25/02/2022 01:52   
3    05/1999  31821770-8E76-434D-BB14-10A6CA8B7667  25/02/2022 01:59   
4    06/2001  D8811E98-FE36-4575-99E5-68BE7D31F6FB  04/04/2022 17:27   

   CodCliente  
0  23498094.0  
1  20236240.0  
2  23413603.0  
3  20078005.0  
4  23821669.0

### 2.5 Merge 04: pesquisa com painelista_pesquisa(merged_03) na chave PesquisaID

In [15]:
merged_04 = pd.merge(pesquisa, merged_03, on='PesquisaID', how='inner')
merged_04.shape

(1200, 10)

In [16]:
merged_04.head()

PesquisaID     Nome  CodJob  \
0  31821770-8E76-434D-BB14-10A6CA8B7667  98DZK9M    4037   
1  31821770-8E76-434D-BB14-10A6CA8B7667  98DZK9M    4037   
2  31821770-8E76-434D-BB14-10A6CA8B7667  98DZK9M    4037   
3  31821770-8E76-434D-BB14-10A6CA8B7667  98DZK9M    4037   
4  31821770-8E76-434D-BB14-10A6CA8B7667  98DZK9M    4037   

                           PainelistaID           Cidade Estado Sexo  \
0  44C12860-704A-4BFC-AA76-46BD5604B60B       Uberlândia     MG    M   
1  46A9177F-BFC7-44A2-8572-545C76FC38E3          Itapeva     SP    F   
2  1E69D5A1-91BC-4966-A2C0-FFB6C3CF52D8  Taboão da Serra     SP    F   
3  E1BCA661-F67D-42FF-92D1-4DDC08B47238  Duque de Caxias     RJ    F   
4  D04CECFD-F857-43A8-8A06-46DC0927A63B            Belém     PA    F   

  Nascimento              Data  CodCliente  
0    09/1994  14/03/2022 17:48  20236240.0  
1    05/1999  25/02/2022 01:59  20078005.0  
2    06/2001  04/04/2022 17:23  20347837.0  
3    12/2004  15/03/2022 17:28  20243661.0  
4    10/1989  05/04/2022 08:42  20361287.0

### 2.6 Merge 05: resposta(merged_df) com painelista_pesquisa(merged_04) na chave PesquisaID

In [17]:
# Encontrando as colunas em comum entre merged_df e merged_04
colunas_comuns = list(set(merged_df.columns).intersection(set(merged_04.columns)))

# Imprimindo as colunas em comum
print("Colunas em comum entre merged_df e merged_XX2:")
print(colunas_comuns)

Colunas em comum entre merged_df e merged_XX2:
['CodJob', 'CodCliente', 'Nome', 'PesquisaID']


In [18]:
merged_Final = pd.merge(merged_df, merged_04, on=colunas_comuns, how='inner')
merged_Final.shape

(5551, 14)

In [19]:
# Visualizar dataframe de forma randomica
merged_Final.sample(5)

CodCliente  CodJob  CodQuestao  \
1983    21633253    4037      122499   
2609    23485055    3794       57627   
2585    23465538    3794       57724   
5113    23833656    3794       57627   
291     20001345    4037      122499   

                                     DescQuestaoPOR  Resposta  \
1983        Para quais finalidades você usa o PIX?          4   
2609                               Você tem filhos?         1   
2585  Você tem alguma atividade remunerada, mensal?         9   
5113                               Você tem filhos?         2   
291         Para quais finalidades você usa o PIX?          2   

                                PesquisaID     Nome  \
1983  31821770-8E76-434D-BB14-10A6CA8B7667  98DZK9M   
2609  D8811E98-FE36-4575-99E5-68BE7D31F6FB  F60KRQX   
2585  D8811E98-FE36-4575-99E5-68BE7D31F6FB  F60KRQX   
5113  D8811E98-FE36-4575-99E5-68BE7D31F6FB  F60KRQX   
291   31821770-8E76-434D-BB14-10A6CA8B7667  98DZK9M   

                         DescOpcaoPOR                          PainelistaID  \
1983  Pagar compras em supermercados   E66CD898-762E-44B1-AC48-BF04E61DA2CA   
2609                              Sim  E412A2CA-F569-48C6-8436-8EF3A28E44DC   
2585     Empregado de Empresa privada  3FE1238D-2CED-44E8-A834-42E802AD43F6   
5113                              Não  02603CC2-687B-461E-8AB3-8ACB39BF53BD   
291                     Pagar contas   D61461B5-3D0F-4439-A3F5-43A2A67D4FBC   

              Cidade Estado Sexo Nascimento              Data  
1983     Nortelândia     MT    F    09/2002  16/07/2022 01:45  
2609     Jaguariaíva     PR    F    02/1990  11/03/2022 20:34  
2585  Angra dos Reis     RJ    M    03/2002  11/03/2022 15:02  
5113       Castanhal     PA    F    07/2002  05/04/2022 07:26  
291          Jacareí     SP    M    11/2005  20/02/2022 16:24

## 3. Antes de entrar propriamente nas respostas, verificar realmente se os tipos estão corretos

In [20]:
merged_Final.dtypes

CodCliente         int64
CodJob             int64
CodQuestao         int32
DescQuestaoPOR    object
Resposta           int32
PesquisaID        object
Nome              object
DescOpcaoPOR      object
PainelistaID      object
Cidade            object
Estado            object
Sexo              object
Nascimento        object
Data              object
dtype: object

Como tem perguntas relacionada a idade, haverá conversão das seguintes colunas: 

In [21]:
# Convertendo a coluna 'Nascimento' para o formato de data e hora, especificando o formato 'mês/ano'
merged_Final['Nascimento'] = pd.to_datetime(merged_Final['Nascimento'], format='%m/%Y')

# Convertendo a coluna 'Data' para o formato de data e hora padrão do pandas
merged_Final['Data'] = pd.to_datetime(merged_Final['Data'])

# Convertendo a coluna 'Sexo' para o tipo de categoria, útil para colunas com um conjunto limitado de valores
merged_Final['Sexo'] = merged_Final['Sexo'].astype('category')

## 4. Respondendo as questões levantadas pelo setor responsável

### a. Qual é a média e desvio padrão de idade dos 600 Painelista? Talvez você precise fazer aproximações para responder esta pergunta. O seu critério de aproximação será avaliado.

In [22]:
# Contar quantos valores únicos existem na coluna 'PainelistaID'
contagem_painelistas = merged_Final['PainelistaID'].nunique()

# Imprimir a contagem
print(f"Quantidade de PainelistaID únicos: {contagem_painelistas}")

Quantidade de PainelistaID únicos: 600


In [23]:
# Definir o dia como 1 para a coluna 'Nascimento' para considerar como 01/MM/YYYY
merged_Final['Nascimento'] = merged_Final['Nascimento'].apply(lambda x: pd.Timestamp(year=x.year, month=x.month, day=1))

# Calcular a idade em anos considerando a data completa da entrevista e do nascimento
merged_Final['Idade'] = ((merged_Final['Data'] - merged_Final['Nascimento']).dt.days / 365.25).astype(int)

In [24]:
# Remover duplicatas baseadas na coluna 'PainelistaID' mantendo a primeira ocorrência
merged_unicos = merged_Final.drop_duplicates(subset='PainelistaID', keep='first')

# Calcular a média e o desvio padrão da idade dos 600 Painelistas únicos
media_idade_600_unicos = merged_unicos['Idade'].mean()
desvio_padrao_idade_600_unicos = merged_unicos['Idade'].std()

# Contar o número de ocorrências únicas de 'PainelistaID'
contagem_unicos = merged_unicos['PainelistaID'].value_counts().reset_index()
contagem_unicos.columns = ['PainelistaID', 'Contagem']

# Imprimir os resultados
print(f"Média de idade dos 600 Painelistas únicos: {media_idade_600_unicos:.2f}")
print(f"Desvio padrão de idade dos 600 Painelistas únicos: {desvio_padrao_idade_600_unicos:.2f}")

Média de idade dos 600 Painelistas únicos: 26.99
Desvio padrão de idade dos 600 Painelistas únicos: 9.38


> Média de idade dos 600 Painelistas únicos: 26.99

> Desvio padrão de idade dos 600 Painelistas únicos: 9.38

> Bônus (Insigth): Saber a idade média e o desvio padrão ajuda muito em entender o público alvo

### b. Quantos % de quem possui filho(s) são do sexo Feminino?

In [25]:
# Filtrar as linhas onde 'CodQuestao' é 57627 (Você tem filhos?), 'Resposta' é 'sim' e 'Sexo' é 'Feminino'
com_filhos_feminino = merged_Final[(merged_Final['CodQuestao'] == 57627)]

# Contar o número de 'PainelistaID' únicos que atendem aos critérios
com_filhos_feminino['PainelistaID'].nunique()

600

In [26]:
# Filtrar as linhas onde 'CodQuestao' é 57627 (Você tem filhos?), 'Resposta' é 'sim' e 'Sexo' é 'Feminino'
com_filhos_feminino = merged_Final[(merged_Final['CodQuestao'] == 57627) & 
                                   (merged_Final['Resposta'] == 1) & 
                                   (merged_Final['Sexo'] == 'F')]

# Contar o número de 'PainelistaID' únicos que atendem aos critérios
contagem_feminino_com_filhos = com_filhos_feminino['PainelistaID'].nunique()

# Calcular a porcentagem em relação ao número total de Painelistas únicos
contagem_painelistas = merged_Final['PainelistaID'].nunique()
percentual_feminino_com_filhos = (contagem_feminino_com_filhos / contagem_painelistas) * 100

# Imprimir os resultados
print(f"Quantidade de mulheres com filhos: {contagem_feminino_com_filhos}")
print(f"Percentual de mulheres com filhos em relação ao total de Painelistas: {percentual_feminino_com_filhos:.2f}%")

Quantidade de mulheres com filhos: 269
Percentual de mulheres com filhos em relação ao total de Painelistas: 44.83%


> Quantidade de mulheres com filhos: 269

> Percentual de mulheres com filhos em relação ao total de Painelistas: 44.83%

> Bônus (Insigth): Compreender a persona de mulheres com um filho, que representa 44,83% dos painelistas, é essencial para tomadas de decisão mais precisas no negócio. Este grupo significativo pode ser uma oportunidade valiosa para estratégias e abordagens personalizadas.

### c. Quantos % dos Painelistas da região SUL possuem renda mensal até R$ 2.999?

In [27]:
# Inicializar DataFrame vazio para armazenar os resultados filtrados
filtro_sul_renda = pd.DataFrame()

# Verificar se 'Estado' é RS, SC ou PR
if merged_Final['Estado'].isin(['RS', 'SC', 'PR']).any():
    
    # Filtrar linhas onde 'CodQuestao' é 57727
    if (merged_Final['CodQuestao'] == 57727).any():
        
        # Aplicar filtro para 'Resposta' ser 1, 2, 3 ou 4
        filtro_sul_renda = merged_Final[(merged_Final['Estado'].isin(['RS', 'SC', 'PR'])) & 
                                        (merged_Final['CodQuestao'] == 57727) &
                                        (merged_Final['Resposta'].isin([1, 2, 3, 4]))]
        
# Contar Painelistas únicos no DataFrame filtrado
total_sul = filtro_sul_renda['PainelistaID'].nunique()

# Calcular a porcentagem em relação ao número total de Painelistas únicos
contagem_painelistas = merged_Final['PainelistaID'].nunique()
percentual = (total_sul / contagem_painelistas) * 100

# Exibir a resposta
print(f"Percentual de Painelistas da região SUL com renda até R$ 2.999: {percentual:.2f}%")

Percentual de Painelistas da região SUL com renda até R$ 2.999: 7.33%


> Percentual de Painelistas da região SUL com renda até R$ 2.999: 7.33%

> Bônus (Insigth): Embora o grupo de painelistas da região Sul com renda até R$ 2.999 represente apenas 7,33% do total, é crucial compreender as personas por região. Seria valioso investigar as porcentagens correspondentes das outras regiões para obter uma visão mais completa. Essa análise mais abrangente pode revelar oportunidades ou desafios significativos que impactam o negócio de forma mais ampla.

### d. Quantos % dos 600 Painelistas usam o Pix para cada uma das finalidades pesquisadas?

In [28]:
# Filtrar linhas onde 'CodQuestao' é 122499 = "Para quais finalidades você usa o PIX?"
total_pix = merged_Final[merged_Final['CodQuestao'] == 122499]

# Contar o número de registros para cada categoria na coluna 'DescOpcaoPOR'
contagem_por_categoria = total_pix['DescOpcaoPOR'].value_counts()

# Somar todos os valores contados
soma_total = contagem_por_categoria.sum()

# Calcular a porcentagem de cada categoria
porcentagem_por_categoria = (contagem_por_categoria / soma_total) * 100

# Formatar a porcentagem para exibir com '%' e duas casas decimais
porcentagem_formatada = porcentagem_por_categoria.map("{:.2f}%".format)

print("Porcentagem de registros por categoria:")
print(porcentagem_formatada)

Porcentagem de registros por categoria:
Transferir $$ para amigos e familiares     25.35%
Pagar contas                               22.82%
Pagar compras em lojas físicas             14.83%
Pagar compras em lojas online              14.05%
Pagar compras em supermercados             12.79%
Pagar prestadores de serviços               9.07%
Outros (Especificar)                        1.08%
Name: DescOpcaoPOR, dtype: object


> Transferir $$ para amigos e familiares     25.35%

> Pagar contas                               22.82%

> Pagar compras em lojas físicas             14.83%

> Pagar compras em lojas online              14.05%

> Pagar compras em supermercados             12.79%

> Pagar prestadores de serviços               9.07%

> Outros (Especificar)                        1.08%

> > Bônus (Insigth): As quatro primeiras categorias representam 77,05% do total. Ao incluir "Pagar compras em supermercados", essa proporção sobe para 89,84%. Seguindo a regra de Pareto 80/20, focar nessas categorias prioritárias, alinhando-as ao objetivo da empresa, pode maximizar o impacto e eficiência das estratégias de negócio.

### e. Quantos % dos Painelistas que usam o Pix para [Transferir $$ para amigos e familiares] OU [Pagar compras em lojas online] são do sexo Masculino?

In [29]:
# Filtrar linhas onde 'CodQuestao' é 122499
filtro_cod_questao = merged_Final[merged_Final['CodQuestao'] == 122499]

if not filtro_cod_questao.empty:
    # Filtrar linhas onde 'Resposta' é 1 ou 5
    filtro_resposta = filtro_cod_questao[(filtro_cod_questao['Resposta'] == 1) | (filtro_cod_questao['Resposta'] == 5)]

    if not filtro_resposta.empty:
        # Filtrar apenas os homens
        filtro_masculino = filtro_resposta[filtro_resposta['Sexo'] == 'M']

        # Contar o número de homens
        total_homens = len(filtro_masculino)

# Remover duplicatas por PainelistaID
filtro_sem_duplicatas = filtro_masculino.drop_duplicates(subset='PainelistaID')

# Contar o número de homens únicos após remover duplicatas
total_homens_unicos = len(filtro_sem_duplicatas)

# Calcular a porcentagem em relação ao número total de Painelistas únicos
percentual_homens_unicos = (total_homens_unicos / contagem_painelistas) * 100

# Exibir os resultados
print(f"Total de homens únicos que responderam 1 ou 5 na questão 122499: {total_homens_unicos}")
print(f"Percentual em relação ao total de Painelistas únicas: {percentual_homens_unicos:.2f}%")

Total de homens únicos que responderam 1 ou 5 na questão 122499: 118
Percentual em relação ao total de Painelistas únicas: 19.67%


> Total de PainelistasID filtrados: 118

> Porcentagem de PainelistasID filtrados em relação ao total: 19.67%

> Bônus (Insigth): A presença de 19,67% dos PainelistasID nestas categorias pode ser vista como uma oportunidade de nicho. Mesmo sendo uma proporção menor, essa concentração sugere segmentos específicos com potencial de engajamento mais profundo. Este foco pode permitir estratégias mais personalizadas e efetivas, maximizando o retorno sobre investimento em iniciativas direcionadas.